modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/milker.dm

```diff
@@ -0,0 +1,298 @@
+	if(user)
+		if(handle_mode++ > HANDLE_MODE_HIGHEST)
+			handle_mode = HANDLE_MODE_LOWEST
+	switch(handle_mode)
+		if(HANDLE_LIGHT)
+			user?.visible_message(
+			message = span_purple("[user] relaxes [user.p_their()] jaw and softens [user.p_their()] milking hand, appearing to opt for a gentler approach."),
+			self_message = span_purple("You relax your jaw and soften your milking hand, opting for a gentler approach."),
+			blind_message = span_purple("You hear a relaxed rustle."),
+			vision_distance = 1)
+			desc = "[initial(desc)]" + "\n[span_purple("It looks somewhat gentle.")]"
+			pleasure_amount = PLEASURE_LIGHT
+			arousal_amount = AROUSAL_LIGHT
+			pain_amount = PAIN_LIGHT
+			squirt_volume = VOLUME_LIGHT
+		if(HANDLE_STRONG)
+			user?.visible_message(
+			message = span_purple("[user] tightens [user.p_their()] jaw and clenches [user.p_their()] milking hand, appearing to opt for a more aggressive approach."),
+			self_message = span_purple("You tighten your jaw and clench your milking hand, opting for a more aggressive approach."),
+			blind_message = span_purple("You hear a vigorous rustle."),
+			vision_distance = 1)
+			desc = "[initial(desc)]" + "\n[span_purple("It looks rather aggressive!")]"
+			pleasure_amount = PLEASURE_STRONG
+			arousal_amount = AROUSAL_STRONG
+			pain_amount = PAIN_STRONG
+			squirt_volume = VOLUME_STRONG
+
+/// Checks if the breasts are present, exposed, lactating, in range, and containing some kind of fluid. Returns a message if not!
+/obj/item/milker/proc/breast_check(mob/living/carbon/being_milked, mob/living/carbon/human/milker, obj/item/organ/genital/breasts/breasts)
+	var/self_suckle = (being_milked == milker) // Grammarize
+	if(!being_milked.client)
+		to_chat(milker, span_warning("[being_milked] is too busy staring off into space to be milked!"))
+		return FALSE
+	if(!being_milked.client || !being_milked.client.prefs.read_preference(/datum/preference/toggle/master_erp_preferences)) // clients are weird
+		to_chat(milker, span_warning("[self_suckle ? "You would prefer to leave those alone!" : "[being_milked] would prefer you leave those alone!"]"))
+		return FALSE
+	if(!being_milked.client || !being_milked.client.prefs.read_preference(/datum/preference/toggle/erp/sex_toy)) // I guess its a sextoy!
+		to_chat(milker, span_warning("[self_suckle ? "You would prefer to put your hands somewhere else!" : "[being_milked] would prefer you to keep your hands to yourself!"]"))
+		return FALSE
+	if(!ishuman(being_milked)) // Trying to milk a robot? A megarachnid? *IAN*?
+		to_chat(milker, span_warning("You can't milk that!"))
+		return FALSE
+	if(being_milked.stat != CONSCIOUS)
+		to_chat(milker, span_warning("[self_suckle ? "You are in no condition to be milked!" : "[being_milked] is in no condition to be milked!"]"))
+		return FALSE
+	if(!istype(breasts))
+		to_chat(milker, span_warning("[self_suckle ? "You don't have any breasts!" : "[being_milked] doesn't seem to have any breasts!"]"))
```

Comment:
This should be at the very start of the proc, as it's kind of one of the most important checks.

Reply:
@Superlagg just this needs doing

Reply:
And, done!